packages/application-generic/src/services/cache/key-builders/entities.ts

```diff
@@ -99,6 +99,21 @@ const buildAuthServiceKey = ({ apiKey }: { apiKey: string }): string =>
     identifierPrefix: IdentifierPrefixEnum.API_KEY,
   });
 
+const buildApiRateLimitKey = ({
+  apiRateLimitCategory,
+  _environmentId,
+}: {
+  apiRateLimitCategory: string;
+  _environmentId: string;
+}): string =>
+  buildCommonKey({
+    type: CacheKeyTypeEnum.ENTITY,
+    keyEntity: CacheKeyPrefixEnum.API_RATE_LIMIT,
+    environmentId: _environmentId,
+    identifierPrefix: IdentifierPrefixEnum.API_RATE_LIMIT_CATEGORY,
+    identifier: apiRateLimitCategory,
+  });
```

Comment:
The cache key generator to store the API rate limit for a given environment and API rate limit category.